proposals/0290-relax-fee-payer-constraint.md

```diff
@@ -0,0 +1,80 @@
+---
+simd: '0290'
+title: Relax Fee Payer Constraint
+authors:
+  - Andrew Fitzgerald (Anza)
+category: Standard
+type: Core
+status: Review
+created: 2025-05-29
+feature:
+---
+
+## Summary
+
+This proposal aims to relax the handling of invalid fee payers in Solana.
+Currently, if a transaction with an invalid fee payer is included in a block,
+the entire block is rejected.
+This proposal suggests that instead of rejecting the entire block, the
+transaction with the invalid fee payer should simply skip execution.
+
+## Motivation
+
+The current constraint forces block-validation to be synchronous since in order
+to determine if a block is valid or not, some subset of transactions must be
+executed in order to determine if the fee payer has sufficient funds to pay the
+transaction fees.
+By relaxing this constraint, we move one step closer towards asynchronous
+validation/execution.
+
+## New Terminology
+
+None.
+
+## Detailed Design
+
+A transaction has a statically determined fee `fee` lamports.
+A transaction can successfully pay fees if:
+
+- The fee payer account has exactly `fee` lamports.
+- The fee payer account has at least X + `rent_exempt_reserve` lamports.
+
+If the fee payer account does not meet either of these conditions, the transaction
+may be included in a block, but it must not be executed.
+The transaction will have no effect on the state.
+
+Invalid fee payer transactions will count their requested,
+or default, compute units (CUs) towards block limits.
```

Comment:
wrt "default", if SIMD-0172 is approved, default will be zero, or a small number (chances are smaller than its would-be actual CUs).